server/schedule/checker/rule_checker.go

```diff
@@ -200,7 +218,7 @@ func (c *RuleChecker) replaceUnexpectRulePeer(region *core.RegionInfo, rf *place
 		c.handleFilterState(region, filterByTempState)
 		return nil, errNoStoreToReplace
 	}
-	newPeer := &metapb.Peer{StoreId: store, Role: rf.Rule.Role.MetaPeerRole(), IsWitness: rf.Rule.IsWitness}
+	newPeer := &metapb.Peer{StoreId: store, Role: rf.Rule.Role.MetaPeerRole(), IsWitness: true}
```

Comment:
Why change it to true?

Reply:
To improve availability, as adding witness is very fast, so the unavailable peer can be quickly removed from members. Afterwards, the witness is promoted or remained unchanged according to the rule.